server/channels/app/platform/websocket_reliable_test.go

```diff
@@ -65,3 +65,50 @@ func TestMarshalDQ(t *testing.T) {
 	assert.Equal(t, 3, dqPtr)
 	assert.Equal(t, events[:3], gotEvents[:3])
 }
+
+func TestUnmarshalDQFullBuffer(t *testing.T) {
+	ps := PlatformService{}
+
+	// Create exactly deadQueueSize events
+	events := make([]*model.WebSocketEvent, deadQueueSize)
+	for i := 0; i < deadQueueSize; i++ {
+		events[i] = model.NewWebSocketEvent(model.WebsocketEventPosted, "t1", "c1", "u1", nil, "").SetSequence(int64(i))
+	}
+
+	// Set up a scenario where the buffer is already filled and has wrapped around
+	// Use index 0 and simulate that the dqPtr has wrapped around to 0 again
+	got, err := ps.marshalDQ(events, 0, 0)
+	require.NoError(t, err)
+	require.Len(t, got, deadQueueSize)
+
+	// Unmarshal the full buffer back
+	gotEvents, dqPtr, err := ps.UnmarshalDQ(got)
+	require.NoError(t, err)
+
+	// Check that dqPtr wraps around to 0, not deadQueueSize
+	assert.Equal(t, 0, dqPtr, "dqPtr should be 0 for a full buffer (deadQueueSize % deadQueueSize = 0)")
+
+	// Verify all events were unmarshaled correctly
+	assert.Equal(t, events, gotEvents)
+
+	// Alternative test: Create a simulation of the circular buffer behavior
+	// This test fills up to the max and ensures wraparound works correctly
+	events2 := make([]*model.WebSocketEvent, deadQueueSize)
+	for i := 0; i < deadQueueSize; i++ {
+		// Create events with sequence numbers that show wraparound
+		// Last event will have highest sequence to demonstrate the break condition
+		// Seq nos: 100 - 228
+		events2[i] = model.NewWebSocketEvent(model.WebsocketEventPosted, "t1", "c1", "u1", nil, "").SetSequence(int64(i + 100))
+	}
+
+	// Marshal only the last entry wrapping to the first to test wraparound detection
+	got2, err := ps.marshalDQ(events2, deadQueueSize-1, 0)
+	require.NoError(t, err)
+	require.Len(t, got2, 1) // Just the last element
+
+	// Unmarshal this single element
+	gotEvents2, dqPtr2, err := ps.UnmarshalDQ(got2)
+	require.NoError(t, err)
+	assert.Equal(t, 1, dqPtr2, "dqPtr should be 1 for a 1-element buffer (1 % deadQueueSize = 1)")
+	assert.Equal(t, events2[deadQueueSize-1], gotEvents2[0])
+}
```

Comment:
Nit: given these are two tests, should we isolate them with two different `t.Run`?

Reply:
ok